securedrop/tests/test_journalist.py

```diff
@@ -53,8 +53,7 @@ def tearDown(self):
     @patch('crypto_util.CryptoUtil.genrandomid',
            side_effect=['bad', VALID_PASSWORD])
     def test_make_password(self, mocked_pw_gen):
-        class fake_config:
-            pass
+        fake_config = SDConfig()
         assert (journalist_app.utils.make_password(fake_config) ==
```

Comment:
replace fake_config with SDConfig

Reply:
I should not, this test was broken in some way. Let me explain it below.

```
>>> class fake_config:
...     pass
... 
>>> type(fake_config)
<type 'classobj'>
```

As you can see above that `fake_cofnig` is actually a variable of type `classobj` (or in other words, it is a class). But, if you read the implementation of `[make_password](https://github.com/freedomofpress/securedrop/blob/develop/securedrop/journalist_app/utils.py#L201)` you will see that actually wants an object in the argument named `config`. Because of the dynamic dance with has/get attr calls, this never failed. But, we should not pass around a `class` variable where our code expects an object with some certain attributes.

This is another good example why we should defined attributes in our code base and should not try to handle things in random dynamic way. Maybe in future we will start adding type annotation to our tests too.

Reply:
Oh, so this actually is a bug fix. It is worth separating bug fixes from refactor (in separate commits). I commented assuming the purpose of the change was to replace fake_config which is the focus of all other changes in this commit, hence my confusion. All good then.

Reply:
>Oh, so this actually is a bug fix. It is worth separating bug fixes from refactor (in separate commits). I commented assuming the purpose of the change was to replace fake_config which is the focus of all other changes in this commit, hence my confusion. All good then.

Correct, this bug fix was not intentional, but just using the new `SDConfig` solved that bug, that is why it is here. I understand your idea about keeping the PRs separate.